pkgs/by-name/jo/joe/macos-fix.patch

```diff
@@ -0,0 +1,11 @@
+diff -Naru i/joe/tty.c w/joe/tty.c
+--- ./joe/tty.c	2018-01-10 17:28:34
++++ ../joe-i-4.6/joe/tty.c	2024-10-02 14:08:22
+@@ -6,6 +6,7 @@
+  *	This file is part of JOE (Joe's Own Editor)
+  */
+ #include "types.h"
++#include <util.h>
```

Comment:
You should enclose this in guards, or else it will break on Linux:

https://github.com/cpredef/predef/blob/master/OperatingSystems.md#macos

```c
#if defined(__APPLE__) && defined(__MACH__)
#include <util.h>
#endif
```

Reply:
this seems to go against your previous review of "not making the patch optional", doesn't it?

Reply:
It doesn't.

As I said before, the patch should be sent to the upstream, because they can incorporate it so that other users outside Nixpkgs (e.g. Homebrew) can benefit from this.

But in the original form the patch can't be applied because it would break on non-Darwin platforms.

In other words, the patch you proposed should be portable. Previous non-Darwin users will not be affected by it, and new Darwin users can use it.

Reply:
Ok, thanks for the explanation. I changed it anyway in the next commit.